widget/embedded/src/pages/LiquiditySourcePage.tsx

```diff
@@ -0,0 +1,152 @@
+import type {
+  LiquiditySourceType,
+  UniqueSwappersGroupType,
+} from '../utils/settings';
+
+import { i18n } from '@lingui/core';
+import {
+  Button,
+  Checkbox,
+  Image,
+  List,
+  ListItemButton,
+  SearchIcon,
+  styled,
+  TextField,
+  Typography,
+} from '@rango-dev/ui';
+import React, { useMemo, useState } from 'react';
+
+import { Layout } from '../components/Layout';
+import { navigationRoutes } from '../constants/navigationRoutes';
+import { useNavigateBack } from '../hooks/useNavigateBack';
+import { useSettingsStore } from '../store/settings';
+import { containsText } from '../utils/numbers';
+import { getUniqueSwappersGroups } from '../utils/settings';
+
+interface PropTypes {
+  supportedSwappers?: string[];
+  sourceType: 'Exchanges' | 'Bridges';
+}
+
+const Container = styled('div', {
+  height: '100%',
+});
+
+const ListContainer = styled('div', {
+  marginTop: '$20',
+  width: '100%',
+  height: '100%',
+  overflow: 'scroll',
+});
+
+export function LiquiditySourcePage({
+  supportedSwappers,
+  sourceType,
+}: PropTypes) {
+  const [searchedFor, setSearchedFor] = useState<string>('');
+  const toggleLiquiditySource = useSettingsStore.use.toggleLiquiditySource();
+  const { navigateBackFrom } = useNavigateBack();
+
+  const supportedUniqueSwappersGroups: Array<UniqueSwappersGroupType> =
+    getUniqueSwappersGroups(supportedSwappers);
+
+  const validTypes: Array<LiquiditySourceType> = [];
+  if (sourceType === 'Exchanges') {
+    validTypes.push('DEX');
+  }
+  if (sourceType === 'Bridges') {
+    validTypes.push('BRIDGE');
+    validTypes.push('AGGREGATOR');
+  }
+
+  const liquiditySources = supportedUniqueSwappersGroups.filter((uniqueItem) =>
+    validTypes.includes(uniqueItem.type)
+  );
+
+  const hasSelectAll = useMemo(
```

Comment:
your memo doesn't add any value here because `liquiditySources` is a reference and on each render it's updating.

Reply:
This comment is also valid for the rest of `useMemo` in this file.